src/posix/client.cpp

```diff
@@ -322,11 +356,33 @@ int main(int argc, char *argv[])
     }
 #endif
 
+    if (config.mTimeout != 0)
+    {
+        end = GetNow() + config.mTimeout;
+    }
+
     while (!isFinished)
     {
-        char   buffer[kLineBufferSize];
-        fd_set readFdSet;
-        int    maxFd = sSessionFd;
+        struct timeval  timeout;
+        struct timeval *timeoutPtr = nullptr;
+        char            buffer[kLineBufferSize];
+        fd_set          readFdSet;
+        int             maxFd = sSessionFd;
+
+        if (config.mTimeout != 0)
+        {
+            uint64_t now = GetNow();
+
+            if (now >= end)
+            {
+                ret = OT_EXIT_SUCCESS;
```

Comment:
Is it ok to use of `OT_EXIT_SUCCESS` (in all cases)?

Reply:
We expect the `ot-ctl` to exit after the given time is passed, rather than waiting for some special messages. The code exits here meets our requirement. So I think it is reasonable to return `OT_EXIT_SUCCESS ` here.